docs/exceptions.md

```diff
@@ -192,13 +239,21 @@ Do:
 * create a custom exception type
 * use `throwM` (`MonadThrow`)
 
+*Comment:* the last bullet in "Do not" is a silly rule and should be struck.
+If the `Left` case is a normal outcome, then leave it be! Making it into an
+exception and throwing it is unambiguously worse. The type is less descriptive,
+and the normal case is mixed in with every truly exceptional case. It's simply
+a lot harder for a caller to use the function properly.
+
 If you want to return `m (Either e a)` from a function, it's
 recommended to define `instance TypeError "NOT AN EXC" => Exception e`
 for your type `e`. If the meaning of `e` type is not related to
 exceptional situations at all, it's not needed to define such
 instance. But for example if `e` denotes a `ParseError`, please do
 define it.
 
+*Comment:* why? What's the value of this?
```

Comment:
The value of adding the `instance TypeError "NOT AN EXC" => Exception e` is that it disallows statically throwing `e` as an `IO` exception. It means that in the following type signature:

```
IO (Either e a)
```

I do not have to handle both the `e` returned in `Either` and the `e` that might have been thrown with `throwIO`.